File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/submodel/metamodel/connected/submodelelement/ConnectedRelationshipElement.java
package org.eclipse.basyx.submodel.metamodel.connected.submodelelement;

import java.util.Map;

import org.eclipse.basyx.submodel.metamodel.api.reference.IReference;
import org.eclipse.basyx.submodel.metamodel.api.submodelelement.relationship.IRelationshipElement;
import org.eclipse.basyx.submodel.metamodel.map.reference.Reference;
import org.eclipse.basyx.submodel.metamodel.map.submodelelement.relationship.RelationshipElement;
import org.eclipse.basyx.vab.modelprovider.VABElementProxy;

/**
 * "Connected" implementation of RelationshipElement
 * 
 * @author rajashek
 *
 */
public class ConnectedRelationshipElement extends ConnectedSubmodelElement implements IRelationshipElement {
	public ConnectedRelationshipElement(VABElementProxy proxy) {
		super(proxy);
	}

	public void setFirst(IReference first) {
		getProxy().setModelPropertyValue(RelationshipElement.FIRST, first);

	}

	@SuppressWarnings("unchecked")
	@Override
	public IReference getFirst() {
		return Reference.createAsFacade((Map<String, Object>) getElem().getPath(RelationshipElement.FIRST));
	}

	public void setSecond(IReference second) {
		getProxy().setModelPropertyValue(RelationshipElement.SECOND, second);

	}

	@SuppressWarnings("unchecked")
	@Override
	public IReference getSecond() {
		return Reference.createAsFacade((Map<String, Object>) getElem().getPath(RelationshipElement.SECOND));
	}
}

File: sdks/c++/basys.sdk.cc/src/submodel/submodel/connected/submodelelement/ConnectedRelationshipElement.cpp
/*
 * ConnectedRelationshipElement.cpp
 *
 *      Author: wendel
 */

#include <BaSyx/submodel/connected/submodelelement/ConnectedRelationshipElement.h>

namespace basyx {
namespace submodel {

ConnectedRelationshipElement::ConnectedRelationshipElement(std::shared_ptr<vab::core::proxy::IVABElementProxy> proxy) :
  ConnectedSubmodelElement(proxy)
{}

void ConnectedRelationshipElement::setFirst(const IReference & first)
{
  //todo
  //this->setProxyValue(submodelelement::RelationshipElementPath::FIRST, first);
}

std::shared_ptr<IReference> ConnectedRelationshipElement::getFirst() const
{
  //todo
  return nullptr;// this->getProxyValue(submodelelement::RelationshipElementPath::FIRST);
}

void ConnectedRelationshipElement::setSecond(const IReference & second)
{
  //todo
  //this->setProxyValue(submodelelement::RelationshipElementPath::SECOND, second);
}

std::shared_ptr<IReference> ConnectedRelationshipElement::getSecond() const
{
  //todo
  return nullptr;// this->getProxyValue(submodelelement::RelationshipElementPath::SECOND);
}

}
}
